projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -394,16 +398,22 @@ export class IgxTreeGridComponent extends IgxGridBaseComponent {
     }
 
     /**
-     * @hidden
+     * Removes the `IgxGridRowComponent` and the corresponding data record by primary key.
+     * Requires that the `primaryKey` property is set.
+     * The method accept rowSelector as a parameter, which is the rowID.
+     * ```typescript
+     * this.grid1.deleteRow(0);
+     * ```
+     * @param rowSelector
+     * @memberof IgxGridBaseComponent
      */
     public deleteRowById(rowId: any) {
-        if (this.transactions.enabled && this.cascadeOnDelete) {
+        if (this.primaryKey && this.foreignKey && this.cascadeOnDelete && this.transactions.enabled) {
```

Comment:
This can be moved into a variable (or 2) since it's repeated below

Reply:
done.